src/hooks/internal/useField.js

```diff
@@ -31,26 +40,33 @@ function useField(componentName, { name, disabled, optional, validate, data }) {
   const errors = getPath(state.errors, name);
   const hasErrors = Array.isArray(errors) && errors.length > 0;
 
+  // we need to store the data in a ref so that if the validateData prop changes, we have a reference
+  // that we can update with the new value.
+  useEffect(() => {
+    fieldDataRef.current = data;
+  }, [data]);
+
   useEffect(() => {
+    const {
+      registerField,
+      unregisterField,
+      name,
+      disabled,
+      optional,
+      validate,
+    } = registerDataRef.current;
+
     registerField(name, {
       disabled,
       optional,
       validate,
-      data,
+      data: fieldDataRef,
```

Comment:
Why not passing `fieldDataRef.current` here so that on the consumer side you won't need to do `.current` everywhere like you do now?